homeassistant/components/epic_games_store/__init__.py

```diff
@@ -0,0 +1,68 @@
+"""The Epic Games Store integration."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import (
+    HomeAssistant,
+    ServiceCall,
+    ServiceResponse,
+    SupportsResponse,
+)
+from homeassistant.util.json import JsonObjectType
+
+from .const import DOMAIN, SERVICE_DATA
+from .coordinator import EGSCalendarUpdateCoordinator
+
+PLATFORMS: list[Platform] = [
+    Platform.CALENDAR,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Epic Games Store from a config entry."""
+
+    coordinator = EGSCalendarUpdateCoordinator(hass, entry)
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    # Services
+    def async_data(call: ServiceCall) -> ServiceResponse:
+        """Handle data service call."""
+        return {
+            **__serialize_data(coordinator, "discount"),
+            **__serialize_data(coordinator, "free"),
+        }
```

Comment:
Add a "type" param ?

Possible values could be "all", "free", "discount". Optional. Defaults to "all"

Return a list of games.